src/plugins/data/server/search/search_service.ts

```diff
@@ -257,7 +249,7 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
           const searchSourceDependencies: SearchSourceDependencies = {
             getConfig: <T = any>(key: string): T => uiSettingsCache[key],
             search: asScoped(request).search,
-            onResponse: (req, res) => shimHitsTotal(res),
+            onResponse: (req, res) => res,
```

Comment:
Do we need this line at all?

Reply:
`onResponse` is a required attribute of `SearchSourceDependencies` and I didn't want to get into a refactor of the way `SearchSource` works so I just left it like this... But honestly that was just being lazy... I can take a look at how involved it is to remove this line.

Reply:
I took a quick look and we could make it optional inside `FetchHandlers` but I'd prefer to leave this for a follow-up PR unless you have objections.